fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/directory/FDBIndexOutput.java

```diff
@@ -216,8 +217,14 @@ private void flush() {
             buffer.flip();
             byte[] arr = new byte[buffer.remaining()];
             buffer.get(arr);
-            flushes.add(fdbDirectory.writeData(id, (int) ( (currentSize - 1) / blockSize), arr));
-            buffer.clear();
+            flushes.add(fdbDirectory.writeData(id, ( (currentSize - 1) / blockSize), arr));
```

Comment:
I think (we've discussed in another context) that these are synchronous calls and should not be treated as async ones. This may help investigate this issue as turning the call to a sync may simplify the code path.

Reply:
I'm not sure that I follow. How is it related to this change?

Reply:
What the FDBDirectory does is to create an Async initiation of a synchronous operation. I am not sure as to why this is done that way since (IIUC) the write to the transaction is immediate (in memory only). In another area I disabled that same exact construct.
Disabling that for now may help in troubleshooting since we won't have multi-threading here anymore.
I also don't mind doing this in another PR.

Reply:
Will do in a subsequent PR